bufferserver/src/main/java/com/datatorrent/bufferserver/internal/PhysicalNode.java

```diff
@@ -71,36 +71,29 @@ public long getUptime()
    * @param d
    * @throws InterruptedException
    */
-  private SerializedData blocker;
-
   public boolean send(SerializedData d)
   {
-    if (d.offset == d.dataOffset) {
-      if (client.write(d.buffer, d.offset, d.length)) {
-        return true;
-      }
-    } else {
-      if (client.send(d.buffer, d.offset, d.length)) {
-        return true;
-      }
+    if (client.send(d.buffer, d.dataOffset, d.length - (d.dataOffset - d.offset))) {
```

Comment:
It is correct. WriteOnlyLengthPrependerClient uses the length of the Slice and writes it directly to the channel (WriteOnlyClient can be used to send raw data). There is no more two scenarios, when length is present and when it is not present.